docs/static/doc-for-plugin.asciidoc

```diff
@@ -0,0 +1,76 @@
+[[plugin-doc]]
+=== Documenting your plugin
+
+Quality documentation with good examples contributes to the adoption of your plugin.
+
+The documentation for your plugin will be generated and published in the
+{logstash-ref}/index.html[Logstash Reference] and the
+{lsplugindocs}[Versioned plugin docs].
+
+[[plugin-doc-file]]
+==== Documentation file
+
+Documentation should be included in your plugin structure as a single file called 'docs/index.asciidoc'.
+
+[[heading-ids]]
+==== Heading IDs
+
+Heading IDs must be in a format that supports generated IDs. This approach
+creates unique IDs when the
+{lsplugindocs}[Versioned plugin docs] are generated.
+
+*Example*
+
+Don't hardcode a heading ID like this: `[[config_models]]`
+
+Instead, define it like this:
+
+[source,txt]
+----------------------------------
+[id="plugins-{type}s-{plugin}-config_models"]
+==== Configuration models
+----------------------------------
+
+If you hardcode an ID, the versioned plugin docs will build correctly the
+first time. The second time the doc build runs, the ID will be flagged as
+a duplicate, and the build will fail.
+
```

Comment:
You should also add a section about links.

Links that point to content in the same file can use angle brackets: `<<plugins-{type}s-{plugin}-config_models>>`.

Links that point to content in other topics in the Logstash Reference need to use the syntax for external links (required for the links to work in the versioned plugin reference): `{logstash-ref}/plugins-codecs-multiline.html[Multiline codec plugin]`. 

Note that we *do* make an exception for links that point to data type descriptions (like `<<boolean, boolean`>>). We have a cleanup step in the conversion script that converts them to the correct syntax.

You'll need to put this into better words, but that's the gist!